--- origsrc/spice-gtk-0.31/src/Makefile.am	2016-02-18 09:29:04.000000000 -0600
+++ src/spice-gtk-0.31/src/Makefile.am	2016-08-04 18:16:21.442369000 -0500
@@ -126,6 +126,10 @@ SPICE_GTK_LIBADD_COMMON =		\
 	$(LIBM)				\
 	$(NULL)
 
+if !OS_WIN32
+SPICE_GTK_LIBADD_COMMON += -lX11
+endif
+
 SPICE_GTK_SOURCES_COMMON =		\
 	glib-compat.h			\
 	gtk-compat.h			\
@@ -585,7 +589,7 @@ $(install_pyexecLTLIBRARIES): install-li
 
 SpiceClientGtk_la_LIBADD = libspice-client-gtk-2.0.la libspice-client-glib-2.0.la $(PYGTK_LIBS)
 SpiceClientGtk_la_CFLAGS = $(GTK_CFLAGS) $(PYTHON_INCLUDES) $(PYGTK_CFLAGS) $(WARN_PYFLAGS)
-SpiceClientGtk_la_LDFLAGS = -module -avoid-version -fPIC
+SpiceClientGtk_la_LDFLAGS = -module -avoid-version -no-undefined -lpython$(PYTHON_VERSION)
 SpiceClientGtk_la_SOURCES = spice-client-gtk-module.c
 nodist_SpiceClientGtk_la_SOURCES = spice-client-gtk-module.defs.c
 
