--- origsrc/spice-gtk-0.7.81-4f76/gtk/Makefile.am	2011-11-15 13:41:46.000000000 -0600
+++ src/spice-gtk-0.7.81-4f76/gtk/Makefile.am	2011-12-19 21:43:53.550497000 -0600
@@ -507,7 +507,7 @@ pyexec_LTLIBRARIES = SpiceClientGtk.la
 
 SpiceClientGtk_la_LIBADD = libspice-client-gtk-2.0.la libspice-client-glib-2.0.la $(PYGTK_LIBS)
 SpiceClientGtk_la_CFLAGS = $(GTK_CFLAGS) $(PYTHON_INCLUDES) $(PYGTK_CFLAGS)
-SpiceClientGtk_la_LDFLAGS = -module -avoid-version -fPIC
+SpiceClientGtk_la_LDFLAGS = -module -avoid-version -no-undefined -lpython$(PYTHON_VERSION)
 SpiceClientGtk_la_SOURCES = spice-client-gtk-module.c
 nodist_SpiceClientGtk_la_SOURCES = spice-client-gtk-module.defs.c
 
