--- origsrc/spice-gtk-0.7.81-4f76/gtk/Makefile.am	2011-11-15 13:41:46.000000000 -0600
+++ src/spice-gtk-0.7.81-4f76/gtk/Makefile.am	2011-12-19 21:43:53.550497000 -0600
@@ -507,7 +507,7 @@ pyexec_LTLIBRARIES = SpiceClientGtk.la
 
 SpiceClientGtk_la_LIBADD = libspice-client-gtk-2.0.la libspice-client-glib-2.0.la $(PYGTK_LIBS)
 SpiceClientGtk_la_CFLAGS = $(GTK_CFLAGS) $(PYTHON_INCLUDES) $(PYGTK_CFLAGS)
-SpiceClientGtk_la_LDFLAGS = -module -avoid-version -fPIC
+SpiceClientGtk_la_LDFLAGS = -module -avoid-version -no-undefined -lpython$(PYTHON_VERSION)
 SpiceClientGtk_la_SOURCES = spice-client-gtk-module.c
 nodist_SpiceClientGtk_la_SOURCES = spice-client-gtk-module.defs.c
 
@@ -515,10 +515,15 @@ CODEGENDIR = `pkg-config --variable=code
 DEFSDIR = `pkg-config --variable=defsdir pygtk-2.0`
 
 spice-client-gtk.defs: $(libspice_client_gtkinclude_HEADERS) $(nodist_libspice_client_gtkinclude_HEADERS) $(libspice_client_glibinclude_HEADERS) $(nodist_libspice_client_glibinclude_HEADERS)
-	$(AM_V_GEN)$(PYTHON) $(CODEGENDIR)/h2def.py -f spice-client-gtk-manual.defs $(libspice_client_gtkinclude_HEADERS) $(nodist_libspice_client_gtkinclude_HEADERS) $(libspice_client_glibinclude_HEADERS) $(nodist_libspice_client_glibinclude_HEADERS) > $@
+	$(AM_V_GEN)$(PYTHON) $(CODEGENDIR)/h2def.py \
+		-f $(srcdir)/spice-client-gtk-manual.defs \
+		$(addprefix $(srcdir)/,$(libspice_client_gtkinclude_HEADERS)) \
+		$(nodist_libspice_client_gtkinclude_HEADERS) \
+		$(addprefix $(srcdir)/,$(libspice_client_glibinclude_HEADERS)) \
+		$(nodist_libspice_client_glibinclude_HEADERS) > $@
 
 spice-client-gtk-module.defs.c: spice-client-gtk.override spice-client-gtk.defs spice-client-gtk-manual.defs
-	@cat $(srcdir)/spice-client-gtk.defs $(srcdir)/spice-client-gtk-manual.defs > tmp.defs
+	@cat spice-client-gtk.defs $(srcdir)/spice-client-gtk-manual.defs > tmp.defs
 	$(AM_V_GEN)pygobject-codegen-2.0 --prefix spice \
 			  --register $(DEFSDIR)/gdk-types.defs \
 			  --register $(DEFSDIR)/gtk-types.defs \
