--- origsrc/spice-gtk-0.8/vapi/Makefile.am	2011-12-22 17:25:57.000000000 -0600
+++ src/spice-gtk-0.8/vapi/Makefile.am	2012-01-26 02:13:06.582836900 -0600
@@ -20,6 +20,7 @@ EXTRA_DIST =						\
 spice-client-glib-2.0.vapi: $(top_builddir)/gtk/SpiceClientGLib-2.0.gir
 	$(AM_V_GEN)$(VAPIGEN)			\
 		--library spice-client-glib-2.0	\
+		--pkg gio-2.0			\
 		$<
 
 spice-client-gtk-$(SPICE_GTK_API_VERSION).vapi: $(top_builddir)/gtk/SpiceClientGtk-$(SPICE_GTK_API_VERSION).gir spice-client-glib-2.0.vapi
